Added BaseNode class with ability to add Units to control who owns it

Praciv opened the pull request:
when a unit reaches the node it will trigger the addUnit function. This will first check if the state is neutral, it will then assign the state to the attacker. If the state is assigned, it will increase or decrease the number of units depending if the state and attacker match. Lastly if the units fall to zero it will reset the state to neutral

Praciv committed:
Added BaseNode class with ability to add Units to control who owns it

when a unit reaches the node it will trigger the addUnit function. This will first check if the state is neutral, it will then assign the state to the attacker. If the state is assigned, it will increase or decrease the number of units depending if the state and attacker match. Lastly if the units fall to zero it will reset the state to neutral

Hanahoa committed:
Added Public and Private folders for C++ code and made an sln file

Hanahoa committed:
Merge branch 'main' into theo_test

Hanahoa reviewed (changes requested):
Good work, it looks really good! There's only a couple of things I want you to add:
1. A health stat so bots have to 'kill' the node to claim it.
2. I don't believe we'll have need for a neutral state for the nodes as they will start as being claimed by the system/anti-virus and the virus/player will have to claim them from it, hence the health stat.
Otherwise, your code looks clean and well written/optimized and I see no other issues with it. I went ahead and updated it to main so you have access to the node meshes and can attach the code to them and hook up the set state function to a timer in blueprints to test it's functionality and get it all hooked up. Kain is finishing up the materials and will add them to the project soon so we can use them instead of the basic test materials I made.

Hanahoa committed:
Updated .cpp and .h file paths

I was dumb and put the .cpp files in public and the .h files in private instead of the other way around.

Praciv committed:
changed basenode to have no neutral state

The node starts either owned by the player or enemy. The unitsOnNode variable acts as the health
